### PR TITLE
Updates For NSKeyedUnarchiver

### PR DESCRIPTION
**Pull request by CuriousTommy**
Seeing how `initForReadingFromData:error:` changes some of the method's behavior. I wanted to make sure I implement some fixes before I start working on `initForReadingFromData:error`.

Let me know if you need me to fix anything.

### CODE REVIEW

File: src/NSKeyedUnarchiver.m [hunk 1]
```diff
@@ -555,14 +555,22 @@ static id _decodeObjectBinary(NSKeyedUnarchiver *unarchiver, NSUInteger uid1) NS
             @throw [NSException exceptionWithName:NSInvalidUnarchiveOperationException reason:[NSString stringWithFormat:@"No classForKeyedUnarchiver class:%@", className] userInfo:nil];
             return nil;
         }
+        NSException *exception = nil;
+        if ([unarchiver requiresSecureCoding])
+        {
+            NSSet *allowed_class_list = [unarchiver allowedClasses];
```
Comment (bugaevc) [inline src/NSKeyedUnarchiver.m]: In objc (well, at least in Cocoa):

* use `NSThisCase` for classes (and other types) and for (typically string or integer) constants
* use `thisCase:andSo:on:` for selectors
* use `thisCase` for variables (local & instance)

Comment (bugaevc) [reply src/NSKeyedUnarchiver.m]: Also this is an `NSSet` so maybe it should be `allowedClassSet`, not list.

Comment (CuriousTommy) [reply src/NSKeyedUnarchiver.m]: Disregard what I said before, I understand what you mean now.

Comment (bugaevc) [reply src/NSKeyedUnarchiver.m]: That is what I'm saying, it's an `NSSet`, so you shouldn't name the variable "allowed class _list_". Maybe just `allowedClasses`, same as the method?

> I grepped for `allowedClassSet`, but I could not find any reference of it in the Foundation repo.

Sorry, I don't understand; why would there be `allowedClassesSet` if you're just adding it?

**Comment by CuriousTommy** [comment]
I can rebase my changes once you are happy with them.

File: src/NSKeyedUnarchiver.m (hunk 1 above)
Comment (CuriousTommy) [reply src/NSKeyedUnarchiver.m]: I was being dumb. I just realized that you were telling me to rename `allowedClassList` to `allowedClassSet`.

**Comment by CuriousTommy** [comment]
@bugaevc Is it okay to merge, or do you want me to make more changes?

**Comment by ahyattdev** [comment]
@bugaevc review?

**Comment by bugaevc** [comment]
LGTM :)

Sorry for not doing anything with this PR; I'm not in my Darling "mode" and haven't been for a while.

We should merge this, update F in the main repo, and I should commit my xtrace work & fixes for building with ninja.

**Comment by ahyattdev** [comment]
@bugaevc sounds great. No worries, I have been pretty busy too. I'll merge this in. The Q2 progress report is being drafted and will be passed around for review before publishing 😄

Thanks for the PR @CuriousTommy!

**Pull request merged**